tests/LanguageServerRename/Handler/Unit/RenameHandlerTest.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServerRename\Tests\Handler\Unit;
+
+use PHPUnit\Framework\TestCase;
+use Phpactor\Extension\LanguageServerBridge\Converter\PositionConverter;
+use Phpactor\Extension\LanguageServerRename\Handler\RenameHandler;
+use Phpactor\Extension\LanguageServerRename\Model\RenameResult;
+use Phpactor\Extension\LanguageServerRename\Model\Renamer;
+use Phpactor\LanguageServerProtocol\Position;
+use Phpactor\LanguageServerProtocol\PrepareRenameParams;
+use Phpactor\LanguageServerProtocol\Range;
+use Phpactor\LanguageServerProtocol\RenameParams;
+use Phpactor\LanguageServerProtocol\TextDocumentEdit;
+use Phpactor\LanguageServerProtocol\TextDocumentIdentifier;
+use Phpactor\LanguageServerProtocol\TextDocumentItem;
+use Phpactor\LanguageServerProtocol\TextEdit as LspTextEdit;
+use Phpactor\LanguageServerProtocol\VersionedTextDocumentIdentifier;
+use Phpactor\LanguageServerProtocol\WorkspaceEdit;
+use Phpactor\LanguageServer\Core\Workspace\Workspace;
+use Phpactor\TextDocument\ByteOffsetRange;
+use Phpactor\TextDocument\TextDocumentBuilder;
+use Phpactor\TextDocument\TextDocumentLocator;
+use Phpactor\TextDocument\TextDocumentUri;
+use Phpactor\TextDocument\TextEdit;
+use Phpactor\TextDocument\TextEdits;
+use Prophecy\Argument;
+use Prophecy\PhpUnit\ProphecyTrait;
+use function Amp\Promise\wait;
+
+class RenameHandlerTest extends TestCase
+{
+    use ProphecyTrait;
+
+    public function testPrepareRename(): void
```

Comment:
This would be better as an integration test .... you could add an additional extension with the `PredefinedRenamer` configured as menitoned above. See the tests for the other handlers.

Reply:
See here for a similar approach https://github.com/phpactor/language-server-phpactor-extensions/blob/35c2564e7fc092bc417bdee230a3c0583caac0b3/tests/LanguageServerIndexer/Extension/TestExtension.php#L14